bagua-core-internal/kernels/bagua_kernels.cu

```diff
@@ -254,6 +254,16 @@ __global__ void divide_inplace_f16(__half *x, float D_, int N) {
     }
 }
 
+__global__ void async_model_average(float *tensor, const float *reduced_tensor_copy, 
+		const float *tensor_copy, const float nranks, const int N) {
+    for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < N; i += blockDim.x * gridDim.x) {
+//        tensor[i] += reduced_tensor_copy[i] / nranks - tensor_copy[i];
+//        if (tensor[i] != tensor[i]) {
+//            printf("nan encountered!");
+//        }
+        atomicAdd(&tensor[i], reduced_tensor_copy[i] / nranks - tensor_copy[i]);
```

Comment:
This is equivalent to "tensor[i] = reduced_tensor_copy[i] / nranks" ?

Reply:
tensor[i] may be different with tensor_copy[i]

since model computation are doing simultaneously with model averaging

Reply:
I see, it's interesting